axelor-budget/src/main/java/com/axelor/apps/budget/service/purchaseorder/PurchaseOrderBudgetServiceImpl.java

```diff
@@ -334,4 +339,25 @@ public void cancelPurchaseOrder(PurchaseOrder purchaseOrder) throws AxelorExcept
       }
     }
   }
+
+  @Override
+  public void autoComputeBudgetDistribution(PurchaseOrder purchaseOrder) throws AxelorException {
+    if (CollectionUtils.isEmpty(purchaseOrder.getPurchaseOrderLineList())
+        || purchaseOrder.getCompany() == null
+        || !budgetToolsService.checkBudgetKeyAndRole(
+            purchaseOrder.getCompany(), AuthUtils.getUser())
+        || !budgetService.checkBudgetKeyInConfig(purchaseOrder.getCompany())) {
+      return;
+    }
```

Comment:
This condition is duplicated every time, maybe it could be a method with a `List<Model>` and a `Company` as parameters (at least a `Company`)
To do for Move, Invoice, PurchaseOrder, SaleOrder